### PR TITLE
scmi-msg drivers

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

**Comment by jforissier** [comment]
- For "drivers: scmi-msg: use if(IS_ENABLED()) rather than #ifdef":
Please add a space between `if` and `(IS_ENABLED())`.
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`
- For "drivers: scmi-msg: fix clock rates query":
`Acked-by: Jerome Forissier <jerome@forissier.org>`

**Comment by etienne-lms** [comment]
Note Travis failed checkaptch `CONFIG_` vs `CFG_` with `IS_ENABLED()` macro.

**Comment by etienne-lms** [comment]
Fixed commit log.
Tags applied.

File: core/drivers/scmi-msg/entry.c [hunk 1]
```diff
@@ -39,16 +40,14 @@ void scmi_process_message(struct scmi_msg *msg)
 	case SCMI_PROTOCOL_ID_BASE:
 		handler = scmi_msg_get_base_handler(msg);
 		break;
-#ifdef CFG_SCMI_MSG_CLOCK
 	case SCMI_PROTOCOL_ID_CLOCK:
-		handler = scmi_msg_get_clock_handler(msg);
+		if (IS_ENABLED(CFG_SCMI_MSG_CLOCK))
+			handler = scmi_msg_get_clock_handler(msg);
```
Comment (etienne-lms) [inline core/drivers/scmi-msg/entry.c:45]: This part is worng. Misses stubs as reported in another P-R https://github.com/OP-TEE/optee_os/pull/4121#discussion_r500238348.

**Comment by etienne-lms** [comment]
Don't merge.
Commit "drivers: scmi-msg: use IS_ENABLED() rather than #ifdef" needs a fixup.

File: core/drivers/scmi-msg/entry.c (hunk 1 above)
Comment (etienne-lms) [reply core/drivers/scmi-msg/entry.c:45]: fixed

**Comment by etienne-lms** [comment]
Actually i was too quick on that. 
The fixup commit is not needed.
The stub function are not needed because here using `IS_ENABLED(CFG_SCMI_MQG_xxx)` to condition function call in **entry.c**, in this P-R.
For consistency, I'd prefer to either remove it, or to keep stubs are remove use of `if (IS_ENABLED(CFG_SCMI_MQG_xxx))` to call the functions.

**Comment by jforissier** [comment]
The usual pattern is to have the stubs so that callers can safely include the header and not be left with unresolved symbols in any case.
Then using IS_ENABLED() or not is a different story, here it does not seem to be required indeed.

File: core/drivers/scmi-msg/entry.c [hunk 2]
```diff
@@ -3,6 +3,7 @@
  * Copyright (c) 2015-2019, Arm Limited and Contributors. All rights reserved.
  * Copyright (c) 2019, Linaro Limited
  */
+#include <config.h>
```
Comment (etienne-lms) [inline core/drivers/scmi-msg/entry.c:6]: to remove

**Comment by etienne-lms** [comment]
Squashed.
Commit log updated for 2nd commit (I removed the review tag).

File: core/drivers/scmi-msg/common.h [hunk 3]
```diff
@@ -104,6 +104,12 @@ scmi_msg_handler_t scmi_msg_get_base_handler(struct scmi_msg *msg);
  * Return a function handler for the message or NULL
  */
 scmi_msg_handler_t scmi_msg_get_clock_handler(struct scmi_msg *msg);
+#else
+static inline scmi_msg_handler_t scmi_msg_get_clock_handler(
+		struct scmi_msg *msg __unused)
```
Comment (etienne-lms) [inline core/drivers/scmi-msg/common.h:109]: checkpatch is not happy here:
```
CHECK: Lines should not end with a '('
#24: FILE: core/drivers/scmi-msg/common.h:108:
+static inline scmi_msg_handler_t scmi_msg_get_clock_handler(
```
Prefer to line break after `inline`?
```
static inline
scmi_msg_handler_t scmi_msg_get_clock_handler(struct scmi_msg *msg __unused)
```

Comment (jforissier) [reply core/drivers/scmi-msg/common.h:109]: > Prefer to line break after `inline`?

Yes, fine with me.

**Comment by jforissier** [comment]
For "drivers: scmi-msg: remove #ifdef in switch/case" with the checkpatch warning fixed:
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

**Comment by etienne-lms** [comment]
checkpatch warning fixed (hopefully) and review tag applied.

**Pull request merged**